crates/criticalup-core/src/binary_proxies.rs

```diff
@@ -46,9 +46,20 @@ pub async fn update(
         "Updating binary proxies"
     );
 
-    let dir = &config.paths.proxies_dir;
-    let list_dir_error = |e| BinaryProxyUpdateError::ListDirectoryFailed(dir.into(), e);
-    match tokio::fs::read_dir(dir).await {
+    let bin_dir = &config.paths.proxy_dir.join("bin");
+    tokio::fs::create_dir_all(bin_dir)
+        .await
+        .map_err(|e| BinaryProxyUpdateError::DirectoryCreationFailed(bin_dir.clone(), e))?;
+    // Required for `rustup toolchain link`
+    tokio::fs::create_dir_all(config.paths.proxy_dir.join("lib"))
+        .await
+        .map_err(|e| BinaryProxyUpdateError::DirectoryCreationFailed(bin_dir.clone(), e))?;
+
+    // Migrate to new proxy system
+    remove_deprecated_proxies(&config.paths.root.join("bin"), &config.paths.proxy_dir).await?;
```

Comment:
Will this be called every time? What if we check if old `bin` exists and then call it?